js/runner_test.go

```diff
@@ -2074,6 +2074,80 @@ func TestMinIterationDurationIsCancellable(t *testing.T) {
 	}
 }
 
+// nolint:paralleltest
+func TestForceHTTP1Feature(t *testing.T) {
+	cases := map[string]struct {
+		godebug               string
+		expectedForceH1Result bool
+		protocol              string
+	}{
+		"Force H1 Enabled. Checking for H1": {
+			godebug:               "http2client=0,gctrace=1",
+			expectedForceH1Result: true,
+			protocol:              "HTTP/1.1",
+		},
+		"Force H1 Disabled. Checking for H2": {
+			godebug:               "test=0",
+			expectedForceH1Result: false,
+			protocol:              "HTTP/2.0",
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			err := os.Setenv("GODEBUG", tc.godebug)
+			require.NoError(t, err)
+			defer func() {
+				err = os.Unsetenv("GODEBUG")
+				require.NoError(t, err)
+			}()
+			assert.Equal(t, tc.expectedForceH1Result, forceHTTP1())
+
+			r1, err := getSimpleRunner(t, "/script.js", fmt.Sprintf(`var k6 = require("k6");
+			var check = k6.check;
+			var fail = k6.fail;
+			var http = require("k6/http");;
+			exports.default = function() {
+				var res = http.get("https://k6.io/");
+				if (
+					!check(res, {
+					'checking to see if status was 200': (res) => res.status === 200,
+					'checking to see if protocol was H2': (res) => res.proto === '%s'
+					})
+				) {
+					fail('test failed')
+				}
+			}`, tc.protocol))
+			require.NoError(t, err)
+
+			err = r1.SetOptions(lib.Options{
```

Comment:
As I asked myself "why?" when seeing it, I think it could be worth adding a comment here documenting why we set the `InsecureSkipTLSVerify` here?